index.js

```diff
@@ -25,13 +25,31 @@ var UNICODE_CHARS = {
     '\u2029': '\\u2029'
 };
 
+// We can‘t just instanceof Date since dates are already converted to strings
+// because of native Date.prototype.JSON (which use toISOString)
+var DATE_LENGTH = new Date().toISOString().length
+function isDate(d) {
+  try {
+    return (
+      // testing length first to avoid new Date() for every string
+      d.length === DATE_LENGTH &&
```

Comment:
this detection process seems to be very flaky, specially because Date() constructor is really flexible. e.g.:

```
new Date("2016-04-29              ")
> Fri Apr 29 2016 00:00:00 GMT-0400 (EDT)
```

Maybe you should use a regexp to detect, then call Date() on the value to validate, and if it throw, you should probably prevent swallowing the error, otherwise this will be crazy to debug.